yt/visualization/plot_window.py

```diff
@@ -2014,3 +2016,121 @@ def SlicePlot(ds, normal=None, fields=None, axis=None, *args, **kwargs):
             del kwargs['north_vector']
 
         return AxisAlignedSlicePlot(ds, normal, fields, *args, **kwargs)
+
+class LinePlot(PlotMPL):
+    r"""
+    A class for constructing line plots
+
+    Parameters
+    ----------
+
+    ds : :class:`yt.data_objects.static_output.Dataset`
+        This is the dataset object corresponding to the
+        simulation output to be plotted.
+    fields : string
+        The name(s) of the field(s) to be plotted.
+    point1: tuple
+        Contains the coordinates of the first point for constructing the line
+    point2: tuple
+        Contains the coordinates of the second point for constructing the line
+    resolution: int
+        How many points to sample between point1 and point2 for constructing
+        the line plot
+    """
+
+    def __init__(self, ds, fields, point1, point2, resolution,
+                 figure_size=5., aspect=None, fontsize=18., labels={}):
+        """
+        Sets up figure and axes
+        """
+        self.handler = ds.coordinates
+        if aspect is None:
+            aspect = 1.
+        fontscale = fontsize / 18.
+        ax_text_size = [1.2*fontscale, 0.9*fontscale]
+        top_buff_size = 0.3*fontscale
+
+        if iterable(figure_size):
+            x_fig_size = figure_size[0]
+            y_fig_size = figure_size[1]
+        else:
+            x_fig_size = figure_size
+            y_fig_size = figure_size/aspect
+
+        x_axis_size = ax_text_size[0]
+        y_axis_size = ax_text_size[1]
+
+        top_buff_size = top_buff_size
+
+        xbins = np.array([x_axis_size, x_fig_size, x_axis_size])
+        ybins = np.array([y_axis_size, y_fig_size, top_buff_size])
+
+        size = [xbins.sum(), ybins.sum()]
+
+        x_frac_widths = xbins/size[0]
+        y_frac_widths = ybins/size[1]
+
+        # axrect is the rectangle defining the area of the
+        # axis object of the plot.  Its range goes from 0 to 1 in
+        # x and y directions.  The first two values are the x,y
+        # start values of the axis object (lower left corner), and the
+        # second two values are the size of the axis object.  To get
+        # the upper right corner, add the first x,y to the second x,y.
+        axrect = (
+            x_frac_widths[0],
+            y_frac_widths[0],
+            x_frac_widths[1],
+            y_frac_widths[1],
+        )
+
+        super(LinePlot, self).__init__(size, axrect, None, None)
+        self.add_plot(fields, point1, point2, resolution, labels=labels)
+        self._xlabel = ("Arc Length [Arb. Units]", 14.)
+        self._ylabel = ("Field Value [Arb. Units]", 14.)
+
+    def add_plot(self, fields, point1, point2, resolution, labels={}):
+        r"""
+        Used to add plots to the figure
+        """
+        if not isinstance(fields, list):
+            fields = [fields]
+        for field in fields:
+            if field not in labels:
+                labels[field] = field[1]
```

Comment:
Ah, I see you convert the start and end points to ndarrays here. You need to make sure these are in the correct units if you are passed a YTArray.

Elsewhere in yt we also accept (array, unit) tuples. Something like:

```
point1 = ([0, 0, 1], 'km')
```

Consider supporting this format as well. We have a number of helpers in the plotting code to help process data that might be an ndarray, generic iterable, YTArray, or (data, unit) tuples. Let me know if you need help finding one to use.